worlds/kh1/Client.py

```diff
@@ -0,0 +1,258 @@
+                    os.remove(root+"/"+file)
+        global item_num
+        item_num = 1
+
+    def on_package(self, cmd: str, args: dict):
+        if cmd in {"Connected"}:
+            if not os.path.exists(self.game_communication_path):
+                os.makedirs(self.game_communication_path)
+            for ss in self.checked_locations:
+                filename = f"send{ss}"
+                with open(os.path.join(self.game_communication_path, filename), 'w') as f:
+                    f.close()
+            
+            #Handle Slot Data
+            for key in list(args['slot_data'].keys()):
+                with open(os.path.join(self.game_communication_path, key + ".cfg"), 'w') as f:
+                    f.write(str(args['slot_data'][key]))
+                    f.close()
+                    
+            ###Support Legacy Games
+            if "Required Reports" in list(args['slot_data'].keys()) and "required_reports_eotw" not in list(args['slot_data'].keys()):
+                reports_required = args['slot_data']["Required Reports"]
+                with open(os.path.join(self.game_communication_path, "required_reports.cfg"), 'w') as f:
+                    f.write(str(reports_required))
+                    f.close()
+            ###End Support Legacy Games
+            
+            #End Handle Slot Data
+
+        if cmd in {"ReceivedItems"}:
+            start_index = args["index"]
+            if start_index != len(self.items_received):
+                global item_num
+                for item in args['items']:
+                    found = False
+                    item_filename = f"AP_{str(item_num)}.item"
+                    for filename in os.listdir(self.game_communication_path):
+                        if filename == item_filename:
+                            found = True
+                    if not found:
+                        with open(os.path.join(self.game_communication_path, item_filename), 'w') as f:
+                            f.write(str(NetworkItem(*item).item) + "\n" + str(NetworkItem(*item).location) + "\n" + str(NetworkItem(*item).player))
+                            f.close()
+                            item_num = item_num + 1
+
+        if cmd in {"RoomUpdate"}:
+            if "checked_locations" in args:
+                for ss in self.checked_locations:
+                    filename = f"send{ss}"
+                    with open(os.path.join(self.game_communication_path, filename), 'w') as f:
+                        f.close()
+
+        if cmd in {"PrintJSON"} and "type" in args:
+            if args["type"] == "ItemSend":
+                item = args["item"]
+                networkItem = NetworkItem(*item)
+                recieverID = args["receiving"]
+                senderID = networkItem.player
+                locationID = networkItem.location
+                if recieverID != self.slot and senderID == self.slot:
+                    itemName = self.item_names[networkItem.item]
```

Comment:
This is a deprecated interface and is not guaranteed to work anymore.
The item ID is no longer enough to know the item name, you have to know both the ID and the player or game.
I think it might look like this:
```python
self.item_names.lookup_in_slot(networkItem.item, networkItem.player)
```
Make sure you test it and don't just trust my guess.

Reply:
Thanks for pointing that out, I had heard about this but hadn't originally written this piece.

Needed to be `itemName = self.item_names.lookup_in_slot(networkItem.item, **recieverID**)`